widgets/cpu_openbsd.lua

```diff
@@ -0,0 +1,60 @@
+-- cpu_openbsd.lua - provide CPU usage on OpenBSD
+-- Copyright (C) 2019  Enric Morales
+--
+-- This file is part of Vicious.
+--
+-- Vicious is free software: you can redistribute it and/or modify
+-- it under the terms of the GNU General Public License as
+-- published by the Free Software Foundation, either version 2 of the
+-- License, or (at your option) any later version.
+--
+-- Vicious is distributed in the hope that it will be useful,
+-- but WITHOUT ANY WARRANTY; without even the implied warranty of
+-- MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+-- GNU Affero General Public License for more details.
+--
+-- You should have received a copy of the GNU Affero General Public License
+-- along with Vicious.  If not, see <https://www.gnu.org/licenses/>.
+
+local math = { ceil = math.ceil }
+local string = { gmatch = string.gmatch }
+local table = { insert = table.insert }
+local tonumber = tonumber
+
+local helpers = require("vicious.helpers")
+
+
+-- cpu_openbsd: provides both a helper function that allows reading
+-- the CPU usage on OpenBSD systems.
+local cpu_openbsd = {}
+
+-- Initialize the table that will contain the ticks spent in each subsystem
+-- values: user, nice, system, spin, interrupts, idle
+cpu_openbsd.ticks = { 0, 0, 0, 0, 0, 0 }
+
+function cpu_openbsd.async(format, warg, callback)
+    helpers.sysctl_async({ "kern.cp_time" },
+	function (ret)
```

Comment:
Notice the mis-indent here.  I believe we need to fix `lua-mode` once and for all (that's the reason I can't switch to 
Emacs completely: it's too difficult to have per language, leave alone per file, config of indentation).  When you're done, please ping me to check out your dotfiles!

Reply:
I am reworking my lua-mode config and will ping you when done.